src/EventListener/StorageListener.php

```diff
@@ -132,15 +136,21 @@ public function postHydrate(HydrationEvent $event)
         }
         $localeData = json_decode($subject[$localeSlug . 'data'], true);
         foreach ($localeData as $key => $value) {
-            if ($key === 'templatefields') {
-                $templateFields = $this->boltConfig->get('theme/templatefields/' . $subject['template'] . '/fields');
-                foreach ($templateFields as $key => $field) {
-                    if ($field['type'] === 'repeater') {
-                        $repeaterData = json_decode($value[$key], true);
-                        /** @var RepeatingFieldCollection[] $subject */
-                        $subject['templatefields'][$key]->clear();
-                        foreach ($repeaterData as $subValue) {
-                            $subject['templatefields'][$key]->addFromArray($subValue);
+            if ($key === 'templatefields' && !( $subject['template']==Null && !isset($contentType['record_template']) )) {
```

Comment:
Strict comparison and using the lowercase `null` constant would be nice, and the spacing is also a bit off (same thing in a few other places below). Something like `!($subject['template'] === null && !isset($contentType['record_template'])`